lib/kubeclient/common.rb

```diff
@@ -427,22 +427,35 @@ def api
     private
 
     def format_response(as, body, list_type = nil)
-      return body if as == :raw
-
-      result = JSON.parse(body)
-
-      if list_type
-        resource_version =
-          result.fetch('resourceVersion') do
-            result.fetch('metadata', {}).fetch('resourceVersion', nil)
-          end
-
-        # result['items'] might be nil due to https://github.com/kubernetes/kubernetes/issues/13096
-        collection = result['items'].to_a.map { |item| Kubeclient::Resource.new(item) }
-
-        Kubeclient::Common::EntityList.new(list_type, resource_version, collection)
+      case as
+      when :raw
+        body
+      when :parsed
+        result = JSON.parse(body)
+        result = result.fetch('items') if list_type
```

Comment:
I'm reluctant to drop information, especially collection resource_version.  "Use this format, unless you need that one info it doesn't support"...
Simplest alternative is just return top-level data, but I agree requiring user to access `[:items]` for common case is cumbersome (and inconsistent with :ros behavior).
How about returning an array that also answers .kind and .resource_version? One way to implement:
```
module EntityListMixin
  attr_accessor :kind, :resource_version
end  

result = JSON.parse(body)
if list_type
  result.extend(EntityListMixin)
  result.kind = list_type
  result.resource_version = resource_version
end
```

P.S. opened #307 to discuss some funkiness in how we set collection `.kind`.  That is orthogonal, but there is one aspect that might be relevant here:
- I'd love `get_foos` in the future to populate .kind on each entity in the array, not just on the collection.  The k8s API is IMHO unfriendly here, it only gives you `"kind": "NodeList"` and assumes you're a strongly-typed client that knows `NodeList.items` is an array of `Node`.

Reply:
why would anyone need the kind/version of something they just queried ?

Reply:
kind is quite redundant but collection resource_version is useful to start a watch from.

Anyway, this can be added later.

Reply:
- `Continue` is never returned from `ros` either ?
 - `Kind` is always what the user requested anyway, so idk if there are a lot of usecases that need that

... changed it back since fetching `items` is easy and means we have less code/feature-requests ...